examples/list-datatype.kl

```diff
@@ -0,0 +1,22 @@
+#lang "n-ary-app.kl"
```

Comment:
To avoid confusion, we should probably rename my `list.kl` example, since it manipulates syntax objects rather than lists.

Reply:
Perhaps syntax-list.kl?

Reply:
Done in 049597f. I did "list-syntax.kl" because it scanned more naturally to me on further thought, but I can do `syntax-list.kl` if you prefer.